library/src/test/java/kwasm/format/text/LimitsTest.kt

```diff
@@ -25,53 +26,59 @@ import org.junit.runners.JUnit4
         val expectedMax = UInt.MAX_VALUE
-        val limits = Type.Limits("123456")
-        Truth.assertThat(limits.value.min).isEqualTo(expectedMin)
-        Truth.assertThat(limits.value.max).isEqualTo(expectedMax)
+        val tokens = tokenizer.tokenize("$expectedMin", context)
+        val parseResult = tokens.parseLimits(0)
+        Truth.assertThat(parseResult.astNode.min).isEqualTo(expectedMin)
+        Truth.assertThat(parseResult.astNode.max).isEqualTo(expectedMax)
     }
 
     @Test
     fun parseMaxVal_setMinToMaxVal() {
         val expectedMin = UInt.MAX_VALUE
         val expectedMax = UInt.MAX_VALUE
-        val limits = Type.Limits(UInt.MAX_VALUE.toString())
-        Truth.assertThat(limits.value.min).isEqualTo(expectedMin)
-        Truth.assertThat(limits.value.max).isEqualTo(expectedMax)
+        val tokens = tokenizer.tokenize("$expectedMin", context)
+        val parseResult = tokens.parseLimits(0)
+        Truth.assertThat(parseResult.astNode.min).isEqualTo(expectedMin)
+        Truth.assertThat(parseResult.astNode.max).isEqualTo(expectedMax)
     }
 
     @Test
     fun parseMinVal_setMinToMinVal() {
         val expectedMin = UInt.MIN_VALUE
         val expectedMax = UInt.MAX_VALUE
-        val limits = Type.Limits(UInt.MIN_VALUE.toString())
-        Truth.assertThat(limits.value.min).isEqualTo(expectedMin)
-        Truth.assertThat(limits.value.max).isEqualTo(expectedMax)
+        val tokens = tokenizer.tokenize("$expectedMin", context)
+        val parseResult = tokens.parseLimits(0)
+        Truth.assertThat(parseResult.astNode.min).isEqualTo(expectedMin)
+        Truth.assertThat(parseResult.astNode.max).isEqualTo(expectedMax)
     }
 
     @Test
     fun parseNegativeMin_throwsParseExceptionWithNegativeNumberMessage() {
-        val limits = Type.Limits("-123456")
-        Assertions.assertThatThrownBy { limits.value.min }
+        val tokens = tokenizer.tokenize("-123456", context)
+        Assertions.assertThatThrownBy { tokens.parseLimits(0) }
             .isInstanceOf(ParseException::class.java)
-            .hasMessageContaining("Illegal char")
+            .hasMessageContaining("Expected integer literal")
     }
 
     @Test
     fun parseABitLargerThanMaxVal_throwsParseExceptionWithValueOverflowMessage() {
-        val limits = Type.Limits("4294967296")
-        Assertions.assertThatThrownBy { limits.value.min }
+        val tokens = tokenizer.tokenize("4294967296", context)
+        Assertions.assertThatThrownBy { tokens.parseLimits(0) }
             .isInstanceOf(ParseException::class.java)
             .hasMessageContaining("Illegal value")
     }
 
     @Test
     fun parseALotLargerThanMaxVal_throwsParseExceptionWithValueOverflowMessage() {
-        val limits = Type.Limits("100000000000")
-        Assertions.assertThatThrownBy { limits.value.min }
+        val tokens = tokenizer.tokenize("100000000000", context)
+        Assertions.assertThatThrownBy { tokens.parseLimits(0) }
```

Comment:
Nit: prefer to import assertThatThrownBy directly

Reply:
Done.

Reply:
Done.